--- a/partner/dpi/reference/drm.cmake
+++ b/partner/dpi/reference/drm.cmake
@@ -10,7 +10,38 @@ include("CheckIncludeFile")
 
 # DRM
 # --------------------------------------------------------------
-if(DPI_REFERENCE_DRM STREQUAL "playready1.2")
+if(DPI_REFERENCE_DRM STREQUAL "playready-dxdrm")
+    set(CMAKE_REQUIRED_INCLUDES_SAVED "${CMAKE_REQUIRED_INCLUDES}")
+    set(CMAKE_REQUIRED_DEFINITIONS_SAVED "${CMAKE_REQUIRED_DEFINITIONS}")
+    set(CMAKE_REQUIRED_INCLUDES "${DRM_INCLUDE}")
+    set(CMAKE_REQUIRED_DEFINITIONS "")
+
+    set(DRM_INCLUDE "${CMAKE_SOURCE_DIR}/../partner/dpi/reference/dxdrm/")
+
+    include_directories(${DRM_INCLUDE})
+
+    if (EXISTS "${DRM_INCLUDE}/external/playready_dxdrm.cpp" )
+        add_definitions( -DPLAYREADY )
+        add_definitions( -DPLAYREADY2 )
+
+#    add_definitions( -DPLAYREADY2_5 )
+#    add_definitions( -DPLAYREADY2TEE )
+#    add_definitions( -DPLAYREADY2_5_TEE )
+#    add_definitions( -DPLAYREADY2_5 )
+#    add_definitions( -DPLAYREADY2_5_TEE_LDL )
+
+        list(APPEND SOURCES ${DRM_INCLUDE}/external/playready_dxdrm.cpp)
+
+        include_directories(${DRM_INCLUDE}/external/)
+
+        list(APPEND LIBRARIES DxDrm provisionproxy)
+#        link_directories(${DRM_LIB})
+    else()
+        message(FATAL_ERROR "playready-dxdrm not properly conconfigured")
+    endif()
+
+
+elseif(DPI_REFERENCE_DRM STREQUAL "playready1.2")
 
     set(PLAYREADY_INCLUDE "${TOOLCHAIN_DIRECTORY}/netflix/include/playready1.2")

--- a/partner/dpi/reference/drm.cmake
+++ b/partner/dpi/reference/drm.cmake
@@ -16,6 +16,7 @@
     set(CMAKE_REQUIRED_INCLUDES "${DRM_INCLUDE}")
     set(CMAKE_REQUIRED_DEFINITIONS "")
 
+#    set(DRM_INCLUDE "${CMAKE_SOURCE_DIR}/../partner/dpi/gstreamer/dxdrm/")
     set(DRM_INCLUDE "${CMAKE_SOURCE_DIR}/../partner/dpi/reference/dxdrm/")
 
     include_directories(${DRM_INCLUDE})
@@ -35,12 +36,11 @@
         include_directories(${DRM_INCLUDE}/external/)
 
         list(APPEND LIBRARIES DxDrm provisionproxy)
-#        link_directories(${DRM_LIB})
+        link_directories(${DRM_LIB})
     else()
-        message(FATAL_ERROR "playready-dxdrm not properly conconfigured")
+        message(FATAL_ERROR "playready-dxdrm not properly configured")
     endif()
 
-
 elseif(DPI_REFERENCE_DRM STREQUAL "playready1.2")
 
     set(PLAYREADY_INCLUDE "${TOOLCHAIN_DIRECTORY}/netflix/include/playready1.2")
--- a/partner/dpi/gstreamer/CMakeLists.txt
+++ b/partner/dpi/gstreamer/CMakeLists.txt
@@ -173,6 +173,8 @@
     list(APPEND SOURCES openmax-il/ComponentOMXIL.cpp)
 endif()
 
+include(drm.cmake)
+
 if(NOT DPI_REFERENCE_DRM STREQUAL "none")
     # Create the DPI directory if we have DRM configured
     set_property(GLOBAL PROPERTY HAVE_DPI_DIRECTORY ${HAVE_DPI_DIRECTORY})
@@ -217,7 +220,7 @@
 
 # DRM
 # --------------------------------------------------------------
-include(drm.cmake)
+#include(drm.cmake)
 
 create_install_target(
     TARGET nrddpi
